cipher/cipher.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2017 Google Inc.
```

Comment:
A few generic comments:
1) The code seems to be AES specific, but looking at the INC, I believe encryption algorithm was supposed to be an option. Is this the first (base) level implementation?
2) How will apid-core get the key/algorithm details?
3)  Can you also add extra comments as to why do the things you do (ex: padding etc), with links so others can easily follow?

Reply:
@sramamoorthy73 
1. Yes. This aims to support KMS data decryption. AES/ECB/PKCS5Padding(PKCS7Padding) is the only supported algorithm for now, because that's how MGMT encrypted.
2. key will be retrieved from MGMT endpoint. algorithm is prepended in encryption data. I'm updating apidVerifyApiKey to add these logics.
3. Currently AES/ECB/PKCS5Padding(PKCS7Padding) is used in MGMT server. I'll ping you with a link of Yehia's MR.

Reply:
Agree additional doc commenting would be great
See this example doc: https://golang.org/src/crypto/cipher/example_test.go (lines 97 - 103)

Sounds like you needed to follow this https://tools.ietf.org/html/rfc5246#section-6.2.3.2
and pad before encrypt